src/tests/infraestructure/controller/TestController.ts

```diff
@@ -0,0 +1,193 @@
+import { Request, Response } from 'express';
+import { services } from '../../../share/infrastructure/services';
+
+export class TestController {
+  async getTests(req: Request, res: Response): Promise<void> {
+    try {
+      const {
+        page = 1,
+        limit = 10,
+        search,
+        companyId,
+        difficulty,
+        orderBy,
+      } = req.query;
+
+      const result = await services.tests.getAll.execute({
+        page: Number(page),
+        limit: Number(limit),
+        search: search as string | undefined,
+        companyId: companyId as string | undefined,
+        difficulty: difficulty as 'EASY' | 'MEDIUM' | 'HARD' | undefined,
+        orderBy: orderBy as string | undefined,
+      });
+
+      res.status(200).json({
+        success: true,
+        ...result,
+      });
+    } catch (error: unknown) {
+      const errorMessage =
+        error instanceof Error ? error.message : 'Unknown error';
+      res.status(500).json({
+        success: false,
+        error: errorMessage,
+      });
+    }
+  }
+
+  async getTestById(req: Request, res: Response): Promise<void> {
+    try {
+      const { id } = req.params;
+
+      if (!id) {
+        res.status(400).json({
+          success: false,
+          error: 'Test ID is required',
+        });
+        return;
+      }
+
+      const result = await services.tests.getById.execute(id);
+      res.status(200).json({
+        success: true,
+        data: result,
+      });
+    } catch (error: unknown) {
+      const errorMessage =
+        error instanceof Error ? error.message : 'Unknown error';
+      const status = errorMessage === 'Test not found' ? 404 : 500;
+      res.status(status).json({
+        success: false,
+        error: errorMessage,
+      });
+    }
+  }
+
+  async startTest(req: Request, res: Response): Promise<void> {
+    try {
+      const { id } = req.params;
+      const { userId } = req.body;
```

Comment:
startTest reads userId from the request body (const { userId } = req.body). Trusting client-provided userId allows impersonation (any user can start sessions on behalf of others).

**Sugerencia:**
```typescript
Derive userId from authentication middleware (e.g. req.user.id after JWT/session validation). Remove userId from the body and enforce server-side auth/ownership checks. Example: const userId = req.user?.id; if (!userId) return 401. Priority: critical (security)
```